tests/test_event_handler.py

```diff
@@ -0,0 +1,22 @@
+from unittest import mock, TestCase
+from src.lib.event_handler import EventHandler
+
+
+class EventHandlerTest(TestCase):
+    @staticmethod
+    def test_all_events_called_on_invoke():
+        my_handler = EventHandler()
+
+        listener_1 = mock.Mock()
+        listener_2 = mock.Mock()
+        listener_3 = mock.Mock()
+
+        my_handler.add(listener_1)
+        my_handler.add(listener_2)
+        my_handler.add(listener_3)
+
+        my_handler.invoke_listeners("foo", bar="baz")
+
+        listener_1.assert_called_once_with("foo", bar="baz")
+        listener_2.assert_called_once_with("foo", bar="baz")
```

Comment:
Includes unit tests! I hope this also helps demonstrate how you would actually use this event listener as well.